tests/Unit/Utilities/Test_StaticCache.cpp

```diff
@@ -43,18 +86,89 @@ SPECTRE_TEST_CASE("Unit.Utilities.StaticCache", "[Utilities][Unit]") {
     ++small_calls;
     return size_t{5};
   });
```

Comment:
I think that right now the documentation doesn't mention that the no-arguments case is also possible... perhaps include a simple case like this one above (but without the counter) as a snippet? Or at least mention it is possible in words.